presto-main/src/main/java/io/prestosql/spiller/AesSpillCipher.java

```diff
@@ -69,7 +69,8 @@ public int encrypt(byte[] data, int inputOffset, int length, byte[] destination,
         Cipher cipher = createEncryptCipher(key);
         System.arraycopy(cipher.getIV(), 0, destination, destinationOffset, ivBytes);
         try {
-            return ivBytes + cipher.doFinal(data, inputOffset, length, destination, destinationOffset + ivBytes);
+            int n = cipher.update(data, inputOffset, length, destination, destinationOffset + ivBytes);
+            return ivBytes + n + cipher.doFinal(destination, destinationOffset + ivBytes + n);
```

Comment:
The gist of the information from the commit message should be also captured as a code comment.
otherwise it's too easy for someone to roll this back into single `doFinal` in a remove "redundant update call" cleanup commit.

Reply:
Done